### ISSUE TITLE
Line 33 of _find.scss: List index is 2 but list is only 1 item long for `nth' on line 33 of helpers/_find.css

### ISSUE THREAD

**Issue by lolmaus** [at-import/Singularity#89]
I've got the following grid definition:

``` sass

// Breakpoints
$s:  400px
$m:  600px
$l:  800px
$xl: 1010px

// Grid for the Featured section
$grids-featured: 151 151
//$grids-featured: 2
$grids-featured: add-grid(151 151 151 151 151 151 at $xl, $grids-featured)
$gutters-featured: 14

// Using the grid (minified example)
$grids:   $grids-featured
$gutters: $gutters-featured

.element:nth-child(1)
  +breakpoint($xl)
    +grid-span(1, 1, $output-style: float)
```

This produces error:

> Line 33 of _find.scss: List index is 2 but list is only 1 item long for `nth'

Trace leads to line 33 of `helpers/_find.css`, then line 4 of `_api.css` (sometimes this step is not in the trace for some reason), then the last line of the above example.

If i change the first grid definition from `151 151` to `2`, the error is gone.

Here's the output of `@warn $grids`.
- erroneous: `151, 151, 151 151 151 151 151 151 1010px`;
- working: `2, 151 151 151 151 151 151 1010px`.

Note the unnecessary comma that appears between elements of the first definition.

The error is caused by this line of code: https://github.com/Team-Sass/Singularity/blob/6e5053adb5216e3923b76b20438eb29a48fc7a0a/stylesheets/singularitygs/gutters/_add.scss#L27

The cause of this problem is a bug in SASS: https://github.com/nex3/sass/issues/706

**Comment by scottkellum** [comment]
Fixed in v1.0.6
Commit: https://github.com/Team-Sass/Singularity/issues/89#ref-commit-24f917e

**Comment by lolmaus** [comment]
Hey Scott, yesterday i've been trying to solve the bug in absolutely the same way: by adding an `@else if` rule in the middle.

The list seems to be assembled correctly now, but another wild error appears:

> Line 35 of _columns.scss: List index is 3 but list is only 2 items long for `nth'

Trace:
- line 35 of .../singularitygs-1.0.6/stylesheets/singularitygs/math/_columns.scss, in `output-float'
- line 4 of .../singularitygs-1.0.6/stylesheets/singularitygs/_api.scss, in `grid-span'
- the call of `grid-span` in the example snippet

**Comment by lolmaus** [comment]
Please reopen.

**Comment by scottkellum** [comment]
I am unable to replicate this error with the code provided. All instances I have are compiling properly. With the example you provided that line doesn’t run for me.

**Comment by lolmaus** [comment]
Roger that. My project is too big to provide full code. I'll try to localize the issue later today or tomorrow.

**Comment by lolmaus** [comment]
My bad!

I had commented out some of my project's grid definitions in order to investigate the bug that you solved with 24f917e6b7318e329bc6eeb8a47606d6ed1f818c. I forgot to uncomment them after updating to the fixed version.

So [my comment from 2013-04-07](#issuecomment-16012420) is a false report, there's no new bug.
